protocol/contracts/interfaces/v2/ITempleStrategyManager.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity 0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Origami (interfaces/v2/ITempleStrategyManager.sol)
+
+// Governs the debt ceilings of each strategy
+interface ITempleStrategyManager {
+
+    struct StrategyConfig {
+        /// @dev Whether this address is enabled as a valid strategy.
+        bool isEnabled;
+
+        /// @dev Governance can pause borrows
+        bool borrowPaused;
+
+        /// @dev Governance can pause repayments
+        bool repaysPaused;
+
+        /// @dev Governance nominates this strategy to be shutdown.
+        /// The strategy then needs to unwind and call shutdown() when ready.
+        bool isShuttingDown;
+
+        /// @dev The total amount of dUSD the strategy can borrow.
+        /// Note: The strategy may end up accruing more debt than this ceiling as it accrues
+        uint256 debtCeiling;
+
+        /// @dev When the strategies totalEquity() is below this amount then operators
+        /// should consider unwinding positions and paying down debt.
+        /// A soft limit for reporting purposes only.
+        uint256 underperformingEquityThreshold;
+    }
```

Comment:
/// @dev Governance nominates this strategy to auto compound into underlying position at select positive Equity threshold.
bool autoCompounding;

/// @ dev when the strategies totalEquity() is above this amount then operators should sell accrued yield and compound into underlying position.
uin256 autocompoundingThreshold;

Reply:
These would make sense for specific strategies - but not all, correct?

I think these parameters could be added just to those strategies where it makes sense (ie each strategy has an individual set of params which could be modified)

Reply:
So you think that each contract should be bespoke or use existing parameters as much as possible?

Reply:
They'll be bespoke parameters for each strategy. It gives flexibility later for the strategy to do what it needs to, and means we won't have to redeploy the manager each time.

It also means the won't be a bunch of non applicable params or params used for something different than what the name of it suggests